hana/lib/drivers/base.js

```diff
@@ -37,20 +37,58 @@ class HANADriver {
         if (streams.length && changes === 0) {
           changes = 1
         }
+
+        if (streams.length) {
+          Promise.all(
+            streams.map(
+              stream =>
+                new Promise((resolve, reject) => {
+                  stream.on('end', () => {
+                    resolve()
+                  })
+                  stream.on('error', () => {
+                    // REVISIT: does it actually matter if it fails or is successful, so should we resolve here?
+                    reject()
+                  })
+                }),
+            ),
+          ).finally(() => {
+            stmt.drop()
+          })
+        }
+
         return { changes }
       },
       runBatch: async params => {
         const stmt = await prep
-        const changes = await prom(stmt, 'exec')(params)
+        const changes = await prom(
+          stmt,
+          'exec',
+        )(params).then(changes => {
+          stmt.drop()
```

Comment:
Calling drop inside these functions is not good. Rather define a new drop function on the returned value and calling it where the `prepare` was originally called. This pattern can also be put into `sqlite` even when it is just an empty function.